hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/impl/TransferPrecompile.java

```diff
@@ -547,6 +547,10 @@ public static CryptoTransferWrapper decodeTransferToken(
         final var receiver = convertLeftPaddedAddressToAccountId(decodedArguments.get(2), aliasResolver, exists);
         final var amount = (long) decodedArguments.get(3);
 
+        if (amount <= 0) {
+            throw new IllegalArgumentException("Amount must be positive");
+        }
```

Comment:
Should we throw for == 0 ?

Reply:
I was debating this myself.  It didn't make sense to me to have a 0 amount transfer.  In the end change it so that we only throw when < 0.